Responsive UI

Samlibardi opened the pull request:
Objetivo: Aumentar a responsividade da UI em telas menores, permitindo a usabilidade do site. Resolver #45 

Alterei o header para utilizar flexbox no arranjo dos itens. O background foi modificado de forma a permitir que a barra aumente verticalmente e mantenha seu design.

Samlibardi committed:
using flexbox so items rearrange on resize

header can now get taller to fit items

Samlibardi committed:
body background set

header background should be applied to the header itself

Samlibardi committed:
fixed header background

Samlibardi committed:
fixed save-controller positioning

Samlibardi committed:
Fixed header width

header now spans through whole page width

Samlibardi committed:
Viewport meta tag

viewport tag to scale correctly on mobile devices

Samlibardi committed:
Main div now flexbox

the scheduler and lecture explorer now reorder if the screen width is small. the explorer goes below the scheduler.
Removed min-width restriction

Samlibardi committed:
Fixed logo positioning

pedropaulovc commented on styles/css/application.css:
```diff
@@ -1,1055 +1 @@
-    visibility: hidden; }
-
-/*# sourceMappingURL=application.css.map */
+html,body,ul,li,header,footer{margin:0;padding:0}ul{list-style:none}li{list-style-type:none}html,body{width:100%;height:100%;min-width:0px;min-height:680px}body{overflow:auto;font-family:tahoma,helvetica;font-size:12px;background:#FEFEFE}#page-container{left:0;right:0;margin-left:auto;margin-right:auto;width:100%;height:100%;max-width:960px;min-width:0px}#main{position:relative;min-height:560px;p ...
```
Apliquei as mudanças no `index.html` e `application.css` no [ambiente de homologação](http://bcc.ime.usp.br/matrusp_homolog/) mas alguma coisa está errada com a renderização no Firefox (58.0b10). O grid de horários não está todo comprimido:

![image](https://user-images.githubusercontent.com/577970/33867331-d7e8916c-deb0-11e7-8692-e790e4f3f23f.png)

Quando este problema for resolvido, eu continuarei com o teste em outros browsers.

pedropaulovc replied:
No Chrome não acontece este problema.

Samlibardi replied:
Interessante. Não havia testado no firefox. Deve ter alguma relação com a altura dos itens dentro da flexbox. Vou tentar resolver e posto o resultado.

pedropaulovc reviewed (changes requested):
Não consegui validar as mudanças no Firefox. Quando este problema for resolvido, eu continuarei com o teste em outros browsers.

Samlibardi committed:
Fixed column height bug, added flex spacer

Columns should be displayed correctly in firefox. Added space between schedule and explorer.

Samlibardi committed:
Added spacer element

Samlibardi committed:
Selected plan highlight improved

Small fix for #31

Samlibardi commented:
Testado e funcionando no firefox 57.0.2

pedropaulovc commented:
Olá Samuel, a UI responsiva está ficando muito boa! Eu validei o MatrUSP no Edge, Chrome e Firefox. Alguns comentários finais antes que eu faça o merge do seu PR:

* O menu para carregar e salvar planos está alinhado com a direita da tela ao invés de ficar à esquerda dos botões 

![image](https://user-images.githubusercontent.com/577970/34026914-1876d2d4-e10e-11e7-9489-dc7e12a8f65b.png)

* No Firefox eu não vejo a barra de likes do Facebook. Não estou muito certo por quê. Edge e Chrome estão funcionando bem.

Samlibardi committed:
fixed header item order, minor tweaks

Samlibardi commented:
Arrumei a ordem dos elementos do header. A barra de likes aparece corretamente no Firefox:
![2017-12-15](https://user-images.githubusercontent.com/29558451/34049953-592d0982-e1a0-11e7-93ad-689922c5a7bb.png)
